clang/lib/CIR/CodeGen/LoweringPrepare.cpp

```diff
@@ -865,30 +893,369 @@ void LoweringPreparePass::lowerThreeWayCmpOp(CmpThreeWayOp op) {
   op.erase();
 }
 
-void LoweringPreparePass::lowerGlobalOp(GlobalOp op) {
-  auto &ctorRegion = op.getCtorRegion();
-  auto &dtorRegion = op.getDtorRegion();
+void LoweringPreparePass::handleGlobalOpCtorDtor(GlobalOp globalOp) {
+  auto &ctorRegion = globalOp.getCtorRegion();
+  auto &dtorRegion = globalOp.getDtorRegion();
 
   if (!ctorRegion.empty() || !dtorRegion.empty()) {
     // Build a variable initialization function and move the initialzation code
     // in the ctor region over.
-    auto f = buildCXXGlobalVarDeclInitFunc(op);
+    auto f = buildCXXGlobalVarDeclInitFunc(globalOp);
 
     // Clear the ctor and dtor region
     ctorRegion.getBlocks().clear();
     dtorRegion.getBlocks().clear();
 
     // Add a function call to the variable initialization function.
     assert(!hasAttr<clang::InitPriorityAttr>(
-               mlir::cast<ASTDeclInterface>(*op.getAst())) &&
+               mlir::cast<ASTDeclInterface>(*globalOp.getAst())) &&
            "custom initialization priority NYI");
     dynamicInitializers.push_back(f);
   }
 
-  std::optional<mlir::ArrayAttr> annotations = op.getAnnotations();
-  if (annotations) {
-    addGlobalAnnotations(op, annotations.value());
+  std::optional<mlir::ArrayAttr> annotations = globalOp.getAnnotations();
+  if (annotations)
+    addGlobalAnnotations(globalOp, annotations.value());
+}
+
+void LoweringPreparePass::lowerGetGlobalOp(GetGlobalOp getGlobalOp) {
```

Comment:
This dance can be simplified. I can just get the static local GlobalOp from the `static_local` `GetGlobalOp`...